src/classes/Entity/types.ts

```diff
@@ -437,39 +438,42 @@ export interface UpdateCustomParameters {
 
 export type UpdateCustomParams = O.Partial<UpdateCustomParameters & DocumentClient.UpdateItemInput>
 
-export type UpdateItem<
-  MethodItemOverlay extends Overlay,
+export type UpdateItem<MethodItemOverlay extends Overlay,
   EntityItemOverlay extends Overlay,
   CompositePrimaryKey extends O.Object,
   Item extends O.Object,
   Attributes extends ParsedAttributes,
-  StrictSchemaCheck extends boolean | undefined = true
-> = FirstDefined<
-  [
+  StrictSchemaCheck extends boolean | undefined = true> = FirstDefined<[
     MethodItemOverlay,
     EntityItemOverlay,
-    A.Compute<
-      CompositePrimaryKey &
-        {
-          [inputAttr in Attributes['always']['input']]:
-            | Item[A.Cast<inputAttr, keyof Item>]
-            | { $delete?: string[]; $add?: any; $prepend?: any[]; $append?: any[] }
-        } &
-        {
-          [optAttr in Attributes['required']['all'] | Attributes['always']['default']]?:
-            | Item[A.Cast<optAttr, keyof Item>]
-            | { $delete?: string[]; $add?: any; $prepend?: any[]; $append?: any[] }
-        } &
-        {
-          [attr in Attributes['optional']]?:
-            | null
-            | Item[A.Cast<attr, keyof Item>]
-            | { $delete?: string[]; $add?: any; $append?: any[]; $prepend?: any[] }
-        } & { $remove?: Attributes['optional'] | Attributes['optional'][] }
-    >
+    A.Compute<CompositePrimaryKey &
+      {
+        [inputAttr in Attributes['always']['input']]:
```

Comment:
We can maybe simplify this with `O.Required` from `ts-toolbelt`:

```typescript
O.Required<
  { [attr in Attributes['always']['input'] | Attributes['required']['all'] | Attributes['always']['default'] | Attributes['optional']]?: ... }, // <= all attributes, maybe this can be simplified as Attributes['all'] ?
  Attributes['always']['input'] // <= marked as required
>
```